DataFormats/PatCandidates/src/Jet.cc

```diff
@@ -249,6 +249,7 @@ void Jet::scaleEnergy(double fScale, const std::string& level) {
   if (jecSetsAvailable()) {
     std::vector<float> factors = {float(jec_[0].correction(0, JetCorrFactors::NONE) / fScale)};
     jec_[0].insertFactor(0, std::make_pair(level, factors));
+    ++currentJECLevel_;
```

Comment:
this method `Jet::scaleEnergy` was added in #27428 with an intention to fix calls like this https://github.com/cms-sw/cmssw/blob/CMSSW_12_1_0/DataFormats/PatCandidates/interface/Jet.h#L407-L410

Now that you are bringing up also possible repeated calls of `scaleEnergy`, I'm not sure that this implementation which always inserts a new factor is going to do the right thing. 

Is the information about the multiple scales needed/required?
- If not, then (somewhat unsafe, but still) assuming that only `scaleEnergy` inserts a `level=0,flavor=NONE` correction at a set 0, should this code instead of repeated insertion accumulate the product?
- Alternatively, if the information about multiple scalings is important then the call to `chargedEmEnergyFraction() const {return chargedEmEnergy() / jecFactor(0) * energy());}` would be incomplete because  `jecFactor(0)` only looks at a set=0, while in this context it should loop over all sets and get a product of `level=0,flavor=NONE` in all available sets

@cms-sw/jetmet-pog-l2 @laurenhay @ahinzmann  please check/clarify.

Reply:
If I understood the second question correctly, I might add a quick comment on it. On [line 250](https://github.com/cms-sw/cmssw/blob/master/DataFormats/PatCandidates/src/Jet.cc#L250) each new scale is added relative to the previous index zero scale. Cumulatively this indicates that ```jecFactor(0) * energy()``` always gives the jet energy where all JECs and all scaleEnergy scales are undone. The functions ```chargedEmEnergy()``` et al. refer to the jet energy composition before all jet energy corrections and scales, so this is sensible.

The first question is open for political discussion. In well-optimized code there should be no need for more than two calls to scaleEnergy, so there is not much difference. There might be some use-cases where both of the scales are needed separately. On the other hand, not so-well optimized code could abuse scaleEnergy calls.

Reply:
> If I understood the second question correctly ...

Thanks for the clarification.
I missed somehow the accumulation of `/fScale` from the previous state.
So, I'd say that no changes are needed here. The first case above is not that essential.